static/app/utils/profiling/profile/continuousProfile.tsx

```diff
@@ -0,0 +1,187 @@
+import type {createContinuousProfileFrameIndex} from 'sentry/utils/profiling/profile/utils';
+
+import {CallTreeNode} from '../callTreeNode';
+import type {Frame} from '../frame';
+
+import {Profile} from './profile';
+
+export class ContinuousProfile extends Profile {
+  static FromChunk(
+    chunk: Profiling.SentryContinousProfileChunk,
+    frameIndex: ReturnType<typeof createContinuousProfileFrameIndex>
+  ): Profile {
+    if (chunk.profile.samples.length <= 1) {
+      return Profile.Empty;
+    }
+
+    const firstSample = chunk.profile.samples[0];
+    const lastSample = chunk.profile.samples[chunk.profile.samples.length - 1];
+
+    const duration = lastSample.timestamp - firstSample.timestamp;
+    const {threadId, threadName} = getThreadData(chunk);
+
+    const profile = new ContinuousProfile({
+      duration,
+      endedAt: lastSample.timestamp,
+      startedAt: firstSample.timestamp,
+      threadId: threadId,
+      name: threadName,
+      type: 'flamechart',
+      unit: 'milliseconds',
+    });
+
+    function resolveFrame(index) {
+      const resolvedFrame = frameIndex[index];
+      if (!resolvedFrame) {
+        throw new Error(`Could not resolve frame ${index} in frame index`);
+      }
+      return resolvedFrame;
+    }
+
+    let frame: Frame | null = null;
+    const resolvedStack: Frame[] = new Array(256); // stack size limit
+
+    for (let i = 0; i < chunk.profile.samples.length; i++) {
+      const sample = chunk.profile.samples[i];
+      const previousSampleTimestamp =
+        chunk.profile.samples[i - 1]?.timestamp ?? firstSample.timestamp;
+
+      const stack = chunk.profile.stacks[sample.stack_id];
+      let size = 0;
+
+      for (let j = 0; j < stack.length; j++) {
+        frame = resolveFrame(stack[j]);
+        if (frame) resolvedStack[size++] = frame;
+      }
+
+      profile.appendSample(
+        resolvedStack,
+        (sample.timestamp - previousSampleTimestamp) * 1e3,
```

Comment:
Don't we normally take the timestamp from the current sample to the _next_ sample for the duration? Here, it's taking the timestamp from the current sample to the _previous_ sample instead.

Reply:
Ah yes, I always get confused by the order